src/Polly.Shared/Wrap/PolicyWrap.cs

```diff
@@ -21,12 +21,12 @@ internal PolicyWrap(Action<Action<Context, CancellationToken>, Context, Cancella
     /// <typeparam name="TResult">The return type of delegates which may be executed through the policy.</typeparam>
     public partial class PolicyWrap<TResult> : Policy<TResult>, IPolicyWrap<TResult>
     {
-        internal PolicyWrap(Func<Func<Context, CancellationToken, TResult>, Context, CancellationToken, TResult> policyAction, Policy outer, IsPolicy inner)
+        internal PolicyWrap(Func<Func<Context, CancellationToken, TResult>, Context, CancellationToken, TResult> policyAction, ISyncPolicy outer, ISyncPolicy<TResult> inner)
             : base(policyAction, outer.ExceptionPredicates,  PredicateHelper<TResult>.EmptyResultPredicates)
```

Comment:
Shouldn't both of these be ISyncPolicy<TResult>?

Reply:
Just like in PolicyWrapAsync.cs, there should be 3 variants:
1: ..., ISyncPolicy outer, ISyncPolicy<TResult> inner)
2: ..., ISyncPolicy<TResult> outer, ISyncPolicy inner)
3: ..., ISyncPolicy<TResult> outer, ISyncPolicy<TResult> inner)
As you mentioned in your next review-comment below, I accidentally forgot to split up 2 and 3, I have pushed the changes to commit b3cc1a7.